presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentStatusUtils.java

```diff
@@ -0,0 +1,146 @@
+import static com.teradata.prestomanager.agent.AgentFileUtils.getFileProperty;
+import static com.teradata.prestomanager.agent.CommandExecutor.executeCommand;
+import static com.teradata.prestomanager.common.RequestBuilder.newRequestBuilder;
+import static java.lang.String.format;
+import static javax.ws.rs.client.Entity.json;
+import static javax.ws.rs.core.Response.Status.OK;
+
+// TODO: stop presto before upgrade and uninstall
+public final class AgentStatusUtils
+{
+    private static final long WAIT_TIME_SECONDS = 60;
+    private static final long STEP_MILLISECONDS = 100;
+    private static final String LAUNCHER_SCRIPT = "/usr/lib/presto/bin/launcher";
+    private static final Path PATH_TO_CONFIG_FILE = Paths.get("/etc/presto/config.properties");
+
+    private AgentStatusUtils() {}
+
+    public static void startPresto()
+            throws PrestoManagerException
+    {
+        if (executeCommand("rpm -q presto-server-rpm") != 0) {
+            throw new PrestoManagerException(format("Presto not installed"));
+        }
+        if (executeCommand("service presto status") == 0) {
+            throw new PrestoManagerException("Presto is already running");
+        }
+        if (executeCommand("service presto start") != 0) {
+            throw new PrestoManagerException("Failed to start presto");
+        }
+        long endTime = System.currentTimeMillis() + TimeUnit.SECONDS.toMillis(WAIT_TIME_SECONDS);
+        while (System.currentTimeMillis() < endTime && !Thread.currentThread().isInterrupted()) {
+            if (executeCommand("service presto status") == 0) {
+                return;
+            }
+            else {
+                sleepUninterruptibly(STEP_MILLISECONDS, TimeUnit.MILLISECONDS);
+            }
+        }
+        throw new PrestoManagerException("Failed to start presto");
+    }
+
+    public static void stopPresto(StopType stopType)
+            throws PrestoManagerException
+    {
+        if (executeCommand("service presto status") != 0) {
+            throw new PrestoManagerException("Presto is not running");
+        }
+        switch (stopType) {
+            case TERMINATE:
+                if (executeCommand("service presto stop") != 0) {
+                    throw new PrestoManagerException("Failed to stop presto");
+                }
+                break;
+            case KILL:
+                executeCommand(format("sudo %s kill", LAUNCHER_SCRIPT));
+                if (executeCommand("service presto status") != 3) {
+                    throw new PrestoManagerException("Failed to stop presto");
+                }
+                break;
+            case GRACEFUL:
+                try {
+                    if (getFileProperty(PATH_TO_CONFIG_FILE, "coordinator").equalsIgnoreCase("true")) {
+                        throw new PrestoManagerException("Coordinator can't be gracefully stopped");
+                    }
```

Comment:
The REST cal would fail if the coordinator is asked to gracefully stop. This check is needed for now I think. We can add extra checks in the controller to prevent this from happening.